.github/workflows/test-action.yml

```diff
@@ -0,0 +1,117 @@
+name: Test Verify Changed Files Action
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ '**' ]
+  workflow_dispatch:
+
+jobs:
+  test-action:
+    runs-on: ubuntu-latest
+    name: Test Verify Changed Files Action
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Full git history for comparisons
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+          cache: 'npm'
+          cache-dependency-path: verify-changed-files/package-lock.json
+
+      - name: Install dependencies
+        run: npm ci
+        working-directory: ./verify-changed-files
+
+      - name: Build Action
+        run: npm run build
+        working-directory: ./verify-changed-files
+
+      - name: Run tests
+        run: npm test
+        working-directory: ./verify-changed-files
+
+      - name: Test Action (All Files)
+        id: verify-all
+        uses: ./verify-changed-files/
+        # No files input - should check entire repository
+        
+      - name: Display all changed files
+        run: |
+          echo "Any changed: ${{ steps.verify-all.outputs.any_changed }}"
+          echo "All changed: ${{ steps.verify-all.outputs.all_changed }}"
+          echo "Changed files: ${{ steps.verify-all.outputs.changed_files }}"
+
+      - name: Test Action (Specific Files)
+        id: verify-specific
+        uses: ./verify-changed-files/
+        with:
+          files: |
+            verify-changed-files/src/**/*.ts
+            verify-changed-files/package.json
+            .github/workflows/test-action.yml
+          
+      - name: Display specific changed files
+        run: |
+          echo "Any specific files changed: ${{ steps.verify-specific.outputs.any_changed }}"
+          echo "All specific files changed: ${{ steps.verify-specific.outputs.all_changed }}"
+          echo "Changed specific files: ${{ steps.verify-specific.outputs.changed_files }}"
+
+      - name: Test Action (Custom Refs)
+        id: verify-custom
+        uses: ./verify-changed-files/
+        with:
+          base-ref: 'HEAD~5'
+          head-ref: 'HEAD'
+          files: 'README.md'
+          
+      - name: Display custom ref changes
+        run: |
+          echo "README changed in last 5 commits: ${{ steps.verify-custom.outputs.any_changed }}"
+
+  verify-dist-updated:
+    runs-on: ubuntu-latest
+    name: Verify dist files are updated
+    if: github.event_name == 'pull_request'
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Full git history for comparisons
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+
+      - name: Check if source files changed
```

Comment:
why do we need to check for sources?

Reply:
This PR level check that verifies if action source code was pushed

Reply:
E.g ensure that dist changes were commited.

Reply:
It does test action with custom configurations (custom path) and also.. prevents action source code being out of sync.

Reply:
If you only check for dist folder after running build you get there, source files will never be changed on CI, am I missing something?

Reply:
This is separate task:

- We do have classic built, test format task (that includes tests for action)
- Check dist task
- Now you have requested separate task for testing actions runtime (use action with assertions in bash etc.)

Reply:
I'm not too sure we are on the same page, my point is you're checking if files changed at `src` or `dist` and I'm saying we should only check `dist`